gravitee-apim-portal-webui-next/src/app/api-details/subscribe-to-api/subscribe-to-api-choose-plan/subscribe-to-api-choose-plan.component.ts

```diff
@@ -0,0 +1,92 @@
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { Component, Input } from '@angular/core';
+import { FormControl, FormGroup, FormsModule, ReactiveFormsModule } from '@angular/forms';
+import { MatButton, MatButtonModule } from '@angular/material/button';
+import { MatCard, MatCardActions, MatCardContent, MatCardHeader } from '@angular/material/card';
+import { MatFormFieldModule } from '@angular/material/form-field';
+import { MatIcon } from '@angular/material/icon';
+import { MatInputModule } from '@angular/material/input';
+import { MatRadioModule } from '@angular/material/radio';
+import { MatStepperModule } from '@angular/material/stepper';
+import { MatTabLink, MatTabNav, MatTabNavPanel } from '@angular/material/tabs';
+import { RouterLinkActive, RouterOutlet } from '@angular/router';
+
+import { BannerComponent } from '../../../../components/banner/banner.component';
+import { BreadcrumbNavigationComponent } from '../../../../components/breadcrumb-navigation/breadcrumb-navigation.component';
+import { PlanCardComponent } from '../../../../components/subscribe/plan-card/plan-card.component';
+import { SubscribeToApiCheckoutComponent } from '../subscribe-to-api-checkout/subscribe-to-api-checkout.component';
+import { SubscribePlan } from '../subscribe-to-api.component';
+
+@Component({
+  selector: 'app-subscribe-to-api-choose-plan',
+  standalone: true,
+  imports: [
+    MatCardContent,
+    PlanCardComponent,
+    MatCardActions,
+    MatCardHeader,
+    BannerComponent,
+    BreadcrumbNavigationComponent,
+    MatButton,
+    MatCard,
+    MatCardContent,
+    MatIcon,
+    MatTabLink,
+    MatTabNav,
+    MatTabNavPanel,
+    RouterLinkActive,
+    RouterOutlet,
+    MatCardHeader,
+    MatRadioModule,
+    PlanCardComponent,
+    MatCardActions,
+    MatButtonModule,
+    MatStepperModule,
+    FormsModule,
+    ReactiveFormsModule,
+    MatFormFieldModule,
+    MatInputModule,
+    SubscribeToApiCheckoutComponent,
+  ],
+  templateUrl: './subscribe-to-api-choose-plan.component.html',
+  styleUrl: './subscribe-to-api-choose-plan.component.scss',
+})
+export class SubscribeToApiChoosePlanComponent {
+  @Input() plans!: SubscribePlan[];
+
+  @Input() subscribeForm!: FormGroup;
+
+  selectedPlan = new FormControl();
+
+  onChangeStepper() {
+    this.subscribeForm.controls['step'].setValue(this.subscribeForm.value.step + this.stepperSelect(<string>this.selectedPlan.value));
+    this.subscribeForm.controls['plan'].setValue(this.getPlan(this.selectedPlan.value));
+  }
+
+  stepperSelect(move: string) {
```

Comment:
The naming here isn't clear, IMHO
How about returning the full number of the next step ?
Or a boolean that says skip next step and you cans create ternary like ` step + skipNextStep ? 2 : 1 ` ?

Reply:
Done